src/fbteken.c

```diff
@@ -1487,13 +1506,19 @@ main(int argc, char *argv[])
 	vtconfigure();
 	drm_backend_show(&gfxstate, &framebuffer);
 
-	winsize.tp_col = framebuffer.width / fnwidth;
-	winsize.tp_row = framebuffer.height / fnheight;
+	if (pivot) {
+		winsize.tp_col = framebuffer.height / fnwidth;
+		winsize.tp_row = framebuffer.width / fnheight;
+	} else {
+		winsize.tp_col = framebuffer.width / fnwidth;
+		winsize.tp_row = framebuffer.height / fnheight;
+	}
 //	winsize.tp_col = 80;
 //	winsize.tp_row = 25;
 	teken_set_winsize(&term.tek, &winsize);
         term.winsz.ws_col = winsize.tp_col;
         term.winsz.ws_row = winsize.tp_row;
+	/* XXX: Should ws_xpixel and ws_ypixel report the physical dimensions or the logical dimensions after pivot? */
```

Comment:
The computation looks fine, we're safer off by reporting logical dimensions (which this code should do, as it is).